File: RepoTransBench/target_projects/Java/C#/code4craft_hello-design-pattern/tests/original/structural/facade/HelloWorldFacadeTest.cs
using Xunit;

namespace HelloDesignPattern.Tests.structural.facade
{
    public class HelloWorldFacadeTest
    {
        [Fact]
        public void TestHelloWorldFacade()
        {
            IHelloWorld facadeHelloWorld = HelloWorldFacade.Instance().FacadeHelloWorld();
            Assert.Equal("Hello Facade!", facadeHelloWorld.HelloWorld());
        }
    }
}

File: RepoTransBench/target_projects/Java/Go/code4craft_hello-design-pattern/tests/original/structural/facade/hello_world_facade_test.go
package facade

import (
	"testing"

	"github.com/stretchr/testify/assert"
)

type HelloWorld interface {
	HelloWorld() string
}

type FacadeHelloWorld struct{}

func (f *FacadeHelloWorld) HelloWorld() string {
	return "Hello Facade!"
}

type HelloWorldFacade struct{}

func NewHelloWorldFacade() *HelloWorldFacade {
	return &HelloWorldFacade{}
}

func (h *HelloWorldFacade) FacadeHelloWorld() HelloWorld {
	return &FacadeHelloWorld{}
}

// Simulate singleton
var facadeInstance = NewHelloWorldFacade()

func Instance() *HelloWorldFacade {
	return facadeInstance
}

func TestHelloWorldFacade(t *testing.T) {
	facadeHelloWorld := Instance().FacadeHelloWorld()
	assert.Equal(t, "Hello Facade!", facadeHelloWorld.HelloWorld())
}